### PR TITLE
Limited GS on Personal: Update plans grid

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/dotcom-forge/issues/2855

## Proposed Changes

Adds the "Style customization" feature to the Personal plan in the plans grid if the user is assigned to the treatment group of the `calypso_global_styles_personal` experiment.

Grid/Flow | Control | Treatment
--- | --- | ---
Main grid | image | Screenshot 2023-07-04 at 15 32 54 | Screenshot 2023-06-26 at 17 46 27 | Screenshot 2023-06-26 at 17 45 13
Newsletter | image| Screenshot 2023-07-04 at 15 34 44
Link in Bio | image | Screenshot 2023-07-04 at 15 34 24
Blog |  | Screenshot 2023-07-05 at 12 07 01

## Testing Instructions

- Use the Calypso live link below.
- Assign yourself to the control group of the experiment (see PCYsg-Fq7-p2#abacus-assignment-bookmarklet and pbxNRc-2HR-p2).
- Run the following in your sandbox to removed the previous cached assignment: `wp_cache_delete( 'global-styles-on-personal-<BLOG_ID>', 'a8c_experiments' );`.
- Go to Upgrades > Plans.
- In the plans main grid, make sure the "Style customization" is included in the Premium plan.
- Click on "Compare plans".
- In the plans comparison grid, make sure the "Style customization" is included in the Premium plan.
- Test the Link in Bio (`/setup/link-in-bio`) or Newsletter (`/setup/newsletter`) flows.
- In the plans step, make sure the "Style customization" is included in the Premium plan.
- Assign yourself to the treatment group of the experiment now.
- Run the following in your sandbox to removed the previous cached assignment: `wp_cache_delete( 'global-styles-on-personal-<BLOG_ID>', 'a8c_experiments' );`.
- Go to Upgrades > Plans.
- In the plans main grid, make sure the "Style customization" is included in the Personal plan.
- Click on "Compare plans".
- In the plans comparison grid, make sure the "Style customization" is included in the Personal plan.
- Test the Link in Bio (`/setup/link-in-bio`) or Newsletter (`/setup/newsletter`) flows.
- In the plans step, make sure the "Style customization" is included in the Personal plan.

### CODE REVIEW

**Review by rcrdortiz — APPROVED** [review]
Great job, everything tests well!

I did find something unrelated to these changes but related to the experiment, we already have it in our minds but just in case. In the link in bio or newsletter flows, the copy is still showing the `style customization` on the Premium plan.

Captura de Pantalla 2023-06-27 a las 10 25 18

My gut also tells me that it should be easier to implement these types of experiments on the plan's list, ideally, we should only need to modify one file, but I couldn't find the best way of doing so (my main idea was to add/remove the feature from the list in the grid index instead of propagating the property across multiple components, but I couldn't get an implementation done, and given that this is a temporal test I think we can live with it)

**Comment by mmtr** [comment]
oh good catch @rcrdortiz. It also affects the main grid:
Screenshot 2023-06-27 at 14 44 31

I think we need to revisit this copy.

**Comment by mmtr** [comment]
Let's wait for feedback about the copy changes before merging this: paYJgx-3wz-p2#comment-3655

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/index.tsx]: We are trying to decouple anything under `plan-features-2023-grid` from Calypso state and other dependencies, and this is just adding more to that work. Can you rework this to pass the value down from `plans-features-main` instead?

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/index.tsx:909]: This looks a little suspect. Why change the type? Sould we also change the type of the AsyncProps passed down from `PlansFeaturesMain`?

**Review by chriskmnds — CHANGES_REQUESTED** [review]
I left some comments that I'd appreciate if you can take a look pls. Thanks

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (chriskmnds) [reply client/my-sites/plan-features-2023-grid/index.tsx:61]: and I just realised we haven't merged the change yet (typed Async props). It's in https://github.com/Automattic/wp-calypso/pull/78731

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (chriskmnds) [reply client/my-sites/plan-features-2023-grid/index.tsx:909]: and I just realised we haven't merged the change yet (typed Async props). It's in https://github.com/Automattic/wp-calypso/pull/78731

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (Copons) [reply client/my-sites/plan-features-2023-grid/index.tsx]: @chriskmnds this should be fixed with dfb3ee0dad9bd24057a801f00d29979582f3e13a, but I'm a bit lost in the flow of the code and all the potential testing venues.
Would you mind confirming it's working as intended when you get the chance?

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (Copons) [reply client/my-sites/plan-features-2023-grid/index.tsx:909]: I presume that whoever merges last will have to revisit this. 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8350876

Thank you @mmtr for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (mmtr) [reply client/my-sites/plan-features-2023-grid/index.tsx:909]: > This looks a little suspect. Why change the type?

Because `PlanFeatures2023GridType` in the intersection `PlanFeatures2023GridProps` and `PlanFeatures2023GridConnectedProps`, and AFAIK `ConnectedPlanFeatures2023Grid` has props from both types.

See https://github.com/Automattic/wp-calypso/blob/1f55d9f821f39b44ad46988a0e97f1fd44eb7cdd/client/my-sites/plan-features-2023-grid/index.tsx#L141-L144.

Also, `isGlobalStylesOnPersonal` is provided by `PlanFeatures2023GridConnectedProps`, not by `PlanFeatures2023GridProps`.

Comment (chriskmnds) [reply client/my-sites/plan-features-2023-grid/index.tsx:909]: Thanks for explaining. Never mind. False alarm - it's the exported grid's type that I'm concerned about, not anything else in this file.

File: client/my-sites/plan-features-2023-grid/index.tsx
Comment (mmtr) [reply client/my-sites/plan-features-2023-grid/index.tsx]: Seems to be working for me, thanks @Copons!

Comment (chriskmnds) [reply client/my-sites/plan-features-2023-grid/index.tsx]: Thank you. All good from my end. also tested a flow and admin and works as outlined.

**Review by chriskmnds — APPROVED** [review]

**Comment by mmtr** [comment]
Copy has been revised (paYJgx-3wz-p2#comment-3694), so @Automattic/t-rex this is ready for review again.

**Review by rcrdortiz — APPROVED** [review]
The changes work as expected, I can see the feature being displayed in the corresponding group based on the experiment group I belong to.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8350876).

**Comment by sirbrillig** [comment]
FTR: Something went weird with the types on this PR which I believe I fixed in https://github.com/Automattic/wp-calypso/pull/79254.

**Pull request merged**
